src/main/java/org/interledger/core/InterledgerAddress.java

```diff
@@ -0,0 +1,70 @@
+package org.interledger.core;
+
+import java.util.Objects;
+
+/**
+ * Interledger Protocol (ILP) Addresses identify Ledger accounts (or groups of Ledger accounts) in an ILP network, and
+ * provide a way to route a payment to its intended destination.
+ * <p>
+ * Interledger Addresses can be subdivided into two categories:
+ * <p>
+ * <b>Destination Addresses</b> are complete addresses that can receive payments. A destination address always maps to
+ * one account in a ledger, though it can also provide more specific information, such as an invoice ID or a
+ * sub-account.
+ * <p>
+ * Destination addresses MUST NOT end in a period (.) character.
+ * <p>
+ * <b>Address Prefixes</b> are incomplete addresses representing a grouping of destination addresses. Many depths of
+ * grouping are possible, for example: groups of accounts or sub-accounts; an individual ledger or subledger; or entire
+ * neighborhoods of ledgers.
+ * <p>
+ * Address prefixes MUST end in a period (.) character.
+ * <p>
+ * The formal specification for an Interledger Addresses is defined in
+ * <a href="https://github.com/interledger/rfcs/tree/master/0015-ilp-addresses">Interledger RFC #15</a>.
+ *
+ * @see "https://github.com/interledger/rfcs/tree/master/0015-ilp-addresses"
+ */
+public interface InterledgerAddress {
```

Comment:
Since we have various teams of people implementing ILP in Java using various types of libraries, this PR proposes to make all of our "core" classes interfaces, so that if somebody doesn't like the community-provided implementation, they would be free to create their own version while being able to utilize it in existing code bases.  Hopefully there's no disagreement here.